code/game/gamemodes/changeling/powers/panacea.dm

```diff
@@ -10,6 +10,12 @@
 /obj/effect/proc_holder/changeling/panacea/sting_action(mob/user)
 	user << "<span class='notice'>We begin cleansing impurities from our form.</span>"
 
+	var/mob/living/simple_animal/borer/borer = user.has_brain_worms()
+	if(borer)
+		borer.leave_victim()
+		if(iscarbon(user))
+			var/mob/living/carbon/C = user
+			C.vomit(0)

```

Comment:
Borers controlling a changeling already can't use any powers, the Changeling tab does not show up for them, Cellular Emporium does though.
But yeah I'll add a check for controlling and a message